src/ai/PathFinder.cpp

```diff
@@ -387,7 +387,7 @@ PathFinder::NodeId PathFinder::getNearestNode(const Vec3f & pos) const {
 	
 	for(size_t i = 0; i < map_s; i++) {
 		float dist = glm::distance2(map_d[i].pos, pos);
-		if(dist < distance && map_d[i].nblinked) {
+		if((dist < distance && map_d[i].nblinked) && !(map_d[i].flags & ANCHOR_FLAG_BLOCKED)) {
```

Comment:
This makes sense but am a bit worried that this might change the resulting NPC behavior. I guess we'll just have to see it causes any problems - should at least be a clear bisect if it does.